To: vim_dev@googlegroups.com
Subject: Patch 7.3.354
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.354
Problem:    ":set backspace+=eol" doesn't work when 'backspace' has a 
            backwards compatible value of 2.
Solution:   Convert the number to a string. (Hirohito Higashi)
Files:      src/option.c


*** ../vim-7.3.353/src/option.c	2011-10-12 16:57:07.000000000 +0200
--- src/option.c	2011-11-30 11:11:15.000000000 +0100
***************
*** 4567,4572 ****
--- 4567,4597 ----
  				arg = errbuf;
  			    }
  			    /*
+ 			     * Convert 'backspace' number to string, for
+ 			     * adding, prepending and removing string.
+ 			     */
+ 			    else if (varp == (char_u *)&p_bs
+ 					 && VIM_ISDIGIT(**(char_u **)varp))
+ 			    {
+ 				i = getdigits((char_u **)varp);
+ 				switch (i)
+ 				{
+ 				    case 0:
+ 					*(char_u **)varp = empty_option;
+ 					break;
+ 				    case 1:
+ 					*(char_u **)varp = vim_strsave(
+ 						      (char_u *)"indent,eol");
+ 					break;
+ 				    case 2:
+ 					*(char_u **)varp = vim_strsave(
+ 						(char_u *)"indent,eol,start");
+ 					break;
+ 				}
+ 				vim_free(oldval);
+ 				oldval = *(char_u **)varp;
+ 			    }
+ 			    /*
  			     * Convert 'whichwrap' number to string, for
  			     * backwards compatibility with Vim 3.0.
  			     * Misuse errbuf[] for the resulting string.
*** ../vim-7.3.353/src/version.c	2011-10-26 23:48:17.000000000 +0200
--- src/version.c	2011-11-30 11:14:44.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     354,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
206. You religiously respond immediately to e-mail, while ignoring
     your growing pile of snail mail.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
